GliaWidgets/ViewModel/Chat/ChatViewModel.swift

```diff
@@ -334,27 +375,54 @@ extension ChatViewModel {
             content: messageText,
             files: files
         )
-        let item = ChatItem(with: outgoingMessage)
-        appendItem(item, to: messagesSection, animated: true)
-        uploader.removeAllUploads()
-        action?(.removeAllUploads)
-        action?(.scrollToBottom(animated: true))
-        let messageTextTemp = messageText
-        messageText = ""
+        
+        if interactor.isEngaged {
+            let item = ChatItem(with: outgoingMessage)
+            appendItem(item, to: messagesSection, animated: true)
+            
+            uploader.removeAllUploads()
+            action?(.removeAllUploads)
+            action?(.scrollToBottom(animated: true))
+            let messageTextTemp = messageText
+
+            interactor.send(messageTextTemp, attachment: attachment) { message in
+                self.replace(
+                    outgoingMessage,
+                    uploads: uploads,
+                    with: message,
+                    in: self.pendingSection
+                )
 
-        interactor.send(messageTextTemp, attachment: attachment) { message in
-            self.replace(
-                outgoingMessage,
-                uploads: uploads,
-                with: message,
-                in: self.messagesSection
+                self.action?(.scrollToBottom(animated: true))
+            } failure: { _ in
```

Comment:
Weak reference is required here, let's add `[weak self]` here.
```suggestion
            } failure: { [weak self] _ in
```